src/main/java/mate/academy/hibernate/relations/dao/impl/MovieDaoImpl.java

```diff
@@ -12,11 +15,38 @@ public MovieDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Movie add(Movie movie) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.persist(movie);
+            transaction.commit();
+            return movie;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Failed to save movie: " + movie);
```

Comment:
In the catch block of the `add` method, you're throwing a `DataProcessingException` without including the original exception `e`. This may make it harder to debug issues since the original stack trace and message are lost. You should include the original exception as a cause when throwing a new exception. For example: `throw new DataProcessingException("Failed to save movie: " + movie, e);`